### PR TITLE
feat: add support for spheres-based functional connectivity

### LINKED ISSUE #41: [MARKER]: Functional Connectivity
### Which marker do you want to include?

No need to convince anyone that it should be possible to compute FC using junifer.

### Is there any publication or available code?

https://nilearn.github.io/stable/auto_examples/03_connectivity/plot_probabilistic_atlas_extraction.html#sphx-glr-auto-examples-03-connectivity-plot-probabilistic-atlas-extraction-py

### Do you have a sample code that implements this outside of junifer?

_No response_

### Anything else to say?

Needs #42 to be implemented first

### PR DESCRIPTION
**Pull request by kaurao**
* [x] fix #41 
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

Spheres based FC

### CODE REVIEW

File: junifer/markers/functional_connectivity_spheres.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    def compute(self, input: Dict, extra_input: Optional[Dict] = None) -> Dict:
+        """Compute.
+
+        Parameters
+        ----------
+        input : Dict[str, Dict]
+            A single input from the pipeline data object in which to compute
+            the marker.
+        extra_input : Dict, optional
```
Comment (synchon) [inline junifer/markers/functional_connectivity_spheres.py:100]: Same as above.

File: junifer/markers/functional_connectivity_spheres.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            The other fields in the pipeline data object. Useful for accessing
+            other data kind that needs to be used in the computation. For
+            example, the functional connectivity markers can make use of the
+            confounds if available (default None).
+
+        Returns
+        -------
+        dict
+            The computed result as dictionary. The following data will be
```
Comment (synchon) [inline junifer/markers/functional_connectivity_spheres.py:109]: We can replace `data` with `keys` to be more explicit.

File: junifer/markers/functional_connectivity_spheres.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        out["kind"] = "tril"
+        return out
+
+    # TODO: complete type annotations
+    def store(self, kind: str, out: Dict, storage) -> None:
+        """Store.
+
+        Parameters
+        ----------
+        input
```
Comment (synchon) [inline junifer/markers/functional_connectivity_spheres.py:147]: Do you want to make changes here for now or fix it once we have the storage interface ready?

File: junifer/markers/tests/test_functional_connectivity_spheres.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+"""Provide test for parcel aggregation."""
```
Comment (synchon) [inline junifer/markers/tests/test_functional_connectivity_spheres.py:1]: Maybe we mention about spheres here?

File: junifer/markers/tests/test_functional_connectivity_spheres.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+# Authors: Amir Omidvarnia <a.omidvarnia@fz-juelich.de>
+#          Kaustubh R. Patil <k.patil@fz-juelich.de>
+#          Federico Raimondo <f.raimondo@fz-juelich.de>
+# License: AGPL
```
Comment (synchon) [inline junifer/markers/tests/test_functional_connectivity_spheres.py:6]: A new line after this.

File: junifer/markers/tests/test_functional_connectivity_spheres.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    assert len(set(out["row_names"])) == 6
+    assert len(set(out["col_names"])) == 6
+
+    # get the timeseries using sa
+    sa = SphereAggregation(
+        coords="DMNBuckner", radius=5.0, method="mean", on="BOLD"
+    )
+    ts = sa.compute({"data": fmri_img})
+
+    # compare with nilearn
```
Comment (synchon) [inline junifer/markers/tests/test_functional_connectivity_spheres.py:48]: Can be removed?

**Review by synchon — CHANGES_REQUESTED** [review]
A few minor changes.

File: junifer/markers/functional_connectivity_spheres.py (hunk 3 above)
Comment (fraimondo) [reply junifer/markers/functional_connectivity_spheres.py:147]: With the storage, we need to fix every docstring

File: junifer/markers/tests/test_functional_connectivity_spheres.py (hunk 6 above)
Comment (fraimondo) [reply junifer/markers/tests/test_functional_connectivity_spheres.py]: fixed some tests

**Comment by fraimondo** [comment]
@synchon your call now

**Review by synchon — APPROVED** [review]
LGTM

**Pull request merged**